Mathlib/Analysis/SpecialFunctions/Complex/CircleAddChar.lean

```diff
@@ -17,9 +17,41 @@ These results are separate from `Analysis.SpecialFunctions.Complex.Circle` in or
 the imports of that file.
 -/
 
-open Complex Function
+open AddChar Complex Function Real
 
-open scoped Real
+namespace Circle
+variable {r s : ℝ}
+
+/-- The standard character sending `r : ℝ` to $$e^{2πir}$$. -/
+noncomputable def e : AddChar ℝ Circle where
```

Comment:
We should relate this def to the ones below, eg toCircle. I think that one is a composition of this and the (not really) canonical map from ZMod N to R given by x -> x/N